application/associations.py

```diff
@@ -263,7 +263,10 @@ def edit_file(self, selection):
 			self._application.create_terminal_tab(active_object._notebook, options)
 
 		else:
-			subprocess.Popen(split_command, cwd=os.path.dirname(selection[0]))
+			cwd = None
```

Comment:
That entirely depends on `.desktop` file definition. If application specifies, for example, `%U` it's requesting URI, instead of local file path. There in lies the problem. Some applications handle URI some don't, but the real problem is properly escaping those.